analyzers/src/SonarAnalyzer.Common/Rules/TooManyLabelsInSwitchBase.cs

```diff
@@ -18,55 +18,50 @@
+    protected const string DiagnosticId = "S1479";
+    private const int DefaultValueMaximum = 30;
 
-        protected abstract DiagnosticDescriptor Rule { get; }
+    protected abstract DiagnosticDescriptor Rule { get; }
 
-        protected abstract TSyntaxKind[] SyntaxKinds { get; }
+    protected abstract TSyntaxKind[] SyntaxKinds { get; }
 
-        protected abstract GeneratedCodeRecognizer GeneratedCodeRecognizer { get; }
+    protected abstract GeneratedCodeRecognizer GeneratedCodeRecognizer { get; }
 
-        protected abstract SyntaxNode GetExpression(TSwitchStatementSyntax statement);
+    protected abstract SyntaxNode GetExpression(TSwitchStatementSyntax statement);
 
-        protected abstract int GetSectionsCount(TSwitchStatementSyntax statement);
+    protected abstract int GetSectionsCount(TSwitchStatementSyntax statement);
 
-        protected abstract Location GetKeywordLocation(TSwitchStatementSyntax statement);
+    protected abstract bool AllSectionsAreOneLiner(TSwitchStatementSyntax statement);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
-            ImmutableArray.Create(Rule);
+    protected abstract Location GetKeywordLocation(TSwitchStatementSyntax statement);
 
-        protected override void Initialize(SonarParametrizedAnalysisContext context)
-        {
-            context.RegisterNodeAction(
-                GeneratedCodeRecognizer,
-                c =>
-                {
-                    var switchNode = (TSwitchStatementSyntax)c.Node;
-                    var type = c.SemanticModel.GetTypeInfo(GetExpression(switchNode)).Type;
+    [RuleParameter("maximum", PropertyType.Integer, "Maximum number of case", DefaultValueMaximum)]
+    public int Maximum { get; set; } = DefaultValueMaximum;
 
-                    if (type == null ||
-                        type.TypeKind == TypeKind.Enum)
-                    {
-                        return;
-                    }
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+        ImmutableArray.Create(Rule);
 
-                    var sectionsCount = GetSectionsCount(switchNode);
-                    if (sectionsCount > Maximum)
-                    {
-                        c.ReportIssue(Rule, GetKeywordLocation(switchNode), Maximum.ToString(), sectionsCount.ToString());
-                    }
-                },
-                SyntaxKinds);
-        }
-    }
+    protected override void Initialize(SonarParametrizedAnalysisContext context) =>
+        context.RegisterNodeAction(
+            GeneratedCodeRecognizer,
+            c =>
+            {
+                var switchNode = (TSwitchStatementSyntax)c.Node;
+
+                if (c.SemanticModel.GetTypeInfo(GetExpression(switchNode)).Type is { TypeKind: not TypeKind.Enum }
+                    && GetSectionsCount(switchNode) > Maximum
+                    && !AllSectionsAreOneLiner(switchNode))
```

Comment:
What about ignoring the onliners in `GetSectionCount` instead?

Reply:
It is still needed to be accounted for the Maximum